detoks/src/main/java/nl/tudelft/trustchain/detoks/Profile.kt

```diff
@@ -3,29 +3,58 @@ package nl.tudelft.trustchain.detoks
 import android.util.Log
 import nl.tudelft.trustchain.detoks.gossiper.NetworkSizeGossiper
 
-/**
- * Basic structure for a profile entry
- */
 class ProfileEntry(
-    var watchTime: Long = 0, // Average watch time
-    val firstSeen: Long = System.currentTimeMillis()
-)
+    var watched:    Boolean = false,
+    val firstSeen:  Long = System.currentTimeMillis(),
+    var uploadDate: Long = 0, // This is the torrent creation date
+    var watchTime:  Long = 0, // Average watch time
+    var duration:   Long = 0, // Video duration
+    var hopCount:   Int  = 0, // Amount of other nodes visited
+    var timesSeen:  Int  = 1, // Count of times we received it
+    var likes:      Int  = 0  // Unimplemented
+) : Comparable<ProfileEntry> {
+    override fun compareTo(other: ProfileEntry): Int = when {
+        this.watchTime != other.watchTime -> this.watchTime compareTo other.watchTime
+        this.firstSeen != other.firstSeen -> this.firstSeen compareTo other.firstSeen
+        else -> 0
+    }
+}
 
 class Profile(
-    val profiles: HashMap<String, ProfileEntry> = HashMap()
+    val torrents: HashMap<String, ProfileEntry> = HashMap()
```

Comment:
Considering profiles are stored, I think that is a more accurate name than torrents